src/crud.js

```diff
@@ -0,0 +1,185 @@
+import dots from '../assets/images/dots.png';
+import removeIcon from '../assets/images/delete.png';
+import saveImg from '../assets/images/save.jpeg';
+
+class Todo {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+const deleteTodos = (e) => {
+  const removeBtn = e.target;
+  const btnClass = removeBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  let existingTodos = JSON.parse(localStorage.getItem('todos'));
+  existingTodos = existingTodos.filter((todos, index) => index !== id);
+  removeBtn.parentNode.remove();
+  existingTodos.forEach((task, i) => (task.index = i + 1));
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+  createTodos();
+};
+
+const clearCompletedTodos = () => {
+  const completed = document.querySelectorAll('input:checked');
+  let existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (completed) {
+    completed.forEach((elem) => {
+      const task = elem.parentNode;
+      task.parentNode.remove();
+    });
+  }
+  existingTodos = existingTodos.filter((todos) => todos.completed === false);
+  existingTodos.forEach((task, i) => (task.index = i + 1));
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
+};
+
+const saveTodos = (e) => {
+  const saveBtn = e.target;
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  const btnClass = saveBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  const taskList = document.querySelector(`#tasks-${id}`);
+  const saveEdit = document.querySelector(`.save-${id}`);
+  const deleteEdit = document.querySelector(`.delete-${id}`);
+  const inputId = document.querySelector(`#complete-${id}`);
+  const editBtn = document.getElementById(`${id}`);
+
+  editBtn.style.display = 'block';
+  inputId.style.outline = 'none';
+
+  existingTodos[id].description = inputId.value;
+  localStorage.setItem('todos', JSON.stringify(existingTodos));
```

Comment:
I can see these two instructions of saving to local storage and reloading the page are usually repeated a lot. what do you think about trying to insert them in a function so that the function gets called once every time for performing those two instructions of saving to local storage and refreshing the page following the DRY methodology and making it well abstracted in a way that if you, for example, want to change the local storage string identifier todos to a different string you only get to change it once and it reflects everywhere you have made an attempt to use it.